### ISSUE TITLE
train: implement lwf_loss regularization function

### ISSUE THREAD

**Issue by BetoBraga** [BetoBraga/continualkit#6]
## Hypothesis
If we add `lwf_loss` to `src/continualkit/train/__init__.py`, the training pipeline has a concrete, testable regularization term that prevents catastrophic forgetting without requiring replay data.

## Concrete Task
Add function `lwf_loss` to `src/continualkit/train/__init__.py`.

```python
def lwf_loss(
    current_logits: "torch.Tensor",
    old_logits: "torch.Tensor",
    temperature: float = 2.0,
) -> "torch.Tensor":
    ...
```

Behavior:
- Implements Learning without Forgetting (LwF) KL-divergence loss
- Softens both distributions with `temperature` before computing KL divergence
- `soft_targets = softmax(old_logits / temperature)`
- `soft_preds = log_softmax(current_logits / temperature)`
- Returns `KLDivLoss(soft_preds, soft_targets) * temperature^2`
- Both tensors must have the same shape — raise `ValueError` if not
- Import torch inside the function body (guard against environments without torch)

## Acceptance Criteria (testable)
- `lwf_loss(logits, logits)` → near 0.0 (identical distributions = no loss)
- `lwf_loss(logits, very_different_logits)` → positive scalar tensor
- Shape mismatch raises `ValueError`
- Higher temperature produces softer targets (lower loss for same inputs)
- Tests use `torch.manual_seed` for reproducibility; mock torch unavailable env
- Tests in `tests/train/test_lwf.py`

## Scope (what NOT to touch)
- Do not implement the full trainer yet — just this one function
- Do not add torch to required dependencies in pyproject.toml (keep it optional, in `train` extra)
- Do not touch anything outside `src/continualkit/train/` and `tests/train/`

## Target Module
src/continualkit/train/
